jobspec2/types.variables.go

```diff
@@ -0,0 +1,691 @@
+package jobspec2
+
+// This file is copied verbatim from Packer: https://github.com/hashicorp/packer/blob/7a1680df97e028c4a75622effe08f6610d0ee5b4/hcl2template/types.variables.go
```

Comment:
Let's go ahead with this, but let's open an issue for how we might be able to work with the Packer folks to pull this common code out somewhere we can share and version between products.